eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/DisallowedWordsCheck.java

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DisallowedWordsCheck extends AbstractCheck {
+    private Set<String> disallowedWords = new HashSet<>(Arrays.asList());
+    private String errorMessage = "%s, All Public API Classes, Fields and Methods should follow " +
+        "Camelcase standards for the following words: %s.";
+
+    /**
+     * Adds words that Classes, Methods and Variables that should follow Camelcasing standards
+     * @param disallowedWords words that should follow normal Camelcasing standards
+     */
+    public final void setDisallowedWords(String... disallowedWords) {
+        if (disallowedWords != null) {
+            Collections.addAll(this.disallowedWords, disallowedWords);
+        }
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.CLASS_DEF,
+        TokenTypes.VARIABLE_DEF,
+        TokenTypes.METHOD_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST token) {
+        switch (token.getType()) {
+            case TokenTypes.CLASS_DEF:
+            case TokenTypes.METHOD_DEF:
+                if (isPublicAPI(token)) {
+                    String tokenName = token.findFirstToken(TokenTypes.IDENT).getText();
+                    boolean found = getDisallowedWords(tokenName);
+                    if (found) {
+                        log(token, String.format(errorMessage, tokenName, this.disallowedWords.stream().collect(Collectors.joining(", ", "", ""))));
+                    }
+                }
+                break;
+            default:
+                // Checkstyle complains if there's no default block in switch
+                break;
+        }
+    }
+
+    /**
+     * Should we check member with given modifiers.
+     *
+     * @param token
+     *                modifiers of member to check.
+     * @return true if we should check such member.
+     */
+    private boolean isPublicAPI(DetailAST token) {
+        final DetailAST modifiersAST =
+            token.findFirstToken(TokenTypes.MODIFIERS);
+        final boolean isStatic = modifiersAST.findFirstToken(TokenTypes.LITERAL_STATIC) != null;
+        final boolean isPublic = modifiersAST
```

Comment:
How about using [TokenUtil.getScopeFromMods](https://checkstyle.org/apidocs/index.html?com/puppycrawl/tools/checkstyle/utils/TokenUtil.html).

```java
final Scope scope = TokenUtil.getScopeFromMods(token);
return scope == Scope.PUBLIC || scope == Scope.PROTECTED) && !isStatic;
```

Reply:
Does not look like the Checkstyle version being used has that method.

Reply:
Looks like it is in now: ScopeUtil.getScopeFromMods(token)